### ISSUE TITLE
PDO integration throws exception when using Propel

### ISSUE THREAD

**Issue by mruoss** [DataDog/dd-trace-php#332]
Propel wraps `PDO` with its own class `PdoConnection`. `PdoConnection` again wraps some calls like `exec()` with its own function and returns an object instead of an int (what `PDO::exec()` returns). 

see 
https://github.com/propelorm/Propel2/blob/master/src/Propel/Runtime/Connection/PdoConnection.php#L115

Now the DDTrace PDO integration traces calls to `PDO::exec()` sets the tag `db.rowcount` to the result of the call. Since the result of `PdoConnection::exec()` returns an object, this step throws an exception because the value passed to `setTag` is an object, not an int.

Is it possible for DDTrace to wrap only the calls to PDO::exec, not the one to `PdoConnection::exec()`?

**Comment by labbati** [comment]
Hi @mruoss and sorry for the late reply. I think this is connected to an issue we have with `parent::` invocation in some specific circumstances and [here it is](https://github.com/propelorm/Propel2/blob/6c4996fc130c7555d82d9dca04396c6f3eeef306/src/Propel/Runtime/Connection/PdoConnection.php#L117).

We are working hard at the moment on this as our top priority and it should be released in the next path release.

We will update this issue once released.

**Comment by SammyK** [comment]
Hey @mruoss! I just wanted to give an update on this. I was able to reproduce your issue and it seems that it is a separate issue from the `parent::` bug. We'll certainly try to get a fix for this on the radar. :)

@pawelchcki & @labbati - Here's a small repro that illustrates the underlying issue:

```php
<?php

class FakePDO
{
    public function exec()
    {
        return 42;
    }
}

class MyPDO extends FakePDO
{
    public function exec()
    {
        $obj = new \stdClass;
        $obj->rows = parent::exec();
        return $obj;
    }
}

dd_trace('FakePDO', 'exec', function() {
    var_dump($this->exec());
});

(new MyPDO())->exec();
```

Since the subclass essentially changes the return type of `exec()` from `int` to `stdClass`, this causes an issue in the integrations (specifically when [setting a tag value](https://github.com/DataDog/dd-trace-php/blob/a8de1d4238a773f03576c8e33fbaf9aebdd64b13/src/DDTrace/Span.php#L186)) when we assume the return values can be cast to strings. So I'm curious your thoughts - perhaps if the value sent to `setTag()` is an object we could check for the existence of the `__toString()` magic method? :)

**Comment by labbati** [comment]
> object we could check for the existence of the __toString() magic method?

This is actually a good point. Not sure this is safe enough though! Object descriptions can be very large and this could cause both space issues and both other problems (e.g.  try to dump a Doctrine entity 😄).

I think that we can safely drop a dump if this is an object. What do you think?

**Comment by SammyK** [comment]
@labbati I like your idea to just ignore the unexpected value. :)

@mruoss I just submitted #423 which should fix this issue. Once the PR is merged and rolled into a release, I'll notify you on this issue. :)

**Comment by mruoss** [comment]
Thanks a lot @SammyK. Looking forward to being able to use this on our project!

**Comment by SammyK** [comment]
Hey @mruoss! We [just released `0.22.0`](https://github.com/DataDog/dd-trace-php/releases/tag/0.22.0) which includes this patch. I'm going to go ahead and close this for now, but let us know if you have any issues with it and I can reopen. Thanks again for reporting this! :)
